rpcs3/Emu/IdManager.h

```diff
@@ -135,7 +136,24 @@ class idm
 	template <typename T>
 	static constexpr u32 get_index(u32 id)
 	{
-		return (id - id_manager::id_traits<T>::base) / id_manager::id_traits<T>::step;
+		using traits = id_manager::id_traits<T>;
+
+		// Note: if id is lower than base, diff / step will be higher than count 
+		u32 diff = id - traits::base;
+
+		if constexpr (traits::step > 1)
```

Comment:
You are doing the compiler's work here, division and modulo by 1 will have similar effect regardless.